cmd/mt-index-cat/out/tpl_pattern_custom.go

```diff
@@ -0,0 +1,132 @@
+package out
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"strings"
+)
+
+type bucket struct {
+	chance int
+	fn     func(in string) string
+}
+
+// percentage chance, and function
+func patternCustom(in ...interface{}) string {
+	usage := func() {
```

Comment:
This is duplicated in flag.Usage().   It should be split out to its own function to use for both?
```
patternCustomUsage(withExample bool) string {
  out := bytes.NewBuffer()
  out.WriteString(fmt.Sprintln("patternCustom: transforms a graphite.style.metric.name into a pattern with wildcards inserted according to rules provided:"))
  ...
  return out.String()
}
```